mint/git-clone/mint-leaf.yml

```diff
@@ -77,13 +80,46 @@ tasks:
 
       git checkout ${{ params.ref }}
 
-      echo "Checked out git repository at $(git rev-parse HEAD)"
+      commit_sha=$(git rev-parse HEAD)
+      echo "Checked out git repository at ${commit_sha}"
 
       if [[ "${{ params.lfs }}" == "true" ]]; then
         git lfs fetch
         git lfs checkout
       fi
 
+      # Set metadata
+      printf "${{ params.repository }}" >> "$MINT_CACHE_KEY_EXCLUDED_ENV/MINT_GIT_REPOSITORY_URL"
+      basename "${{ params.repository }}" | sed 's/\.git$//' >> "$MINT_CACHE_KEY_EXCLUDED_ENV/MINT_GIT_REPOSITORY_NAME"
+      git log -n 1 --pretty=format:%B >> "$MINT_CACHE_KEY_EXCLUDED_ENV/MINT_GIT_COMMIT_MESSAGE"
+      printf "${commit_sha}" >> "$MINT_CACHE_KEY_EXCLUDED_ENV/MINT_GIT_COMMIT_SHA"
+
+      unresolved_ref=""
+      if [[ -n "${{ params.meta-ref }}" ]]; then
+        refs_matching_provided_ref=$(git for-each-ref --format="%(refname)" "refs/heads/${{ params.meta-ref }}" "refs/tags/${{ params.meta-ref }}" "${{ params.meta-ref }}" | grep -v refs/remotes)
+        unresolved_ref=$(echo "$refs_matching_provided_ref" | head -n 1 | tr -d '\n')
+
+        # also, ensure the meta-ref contains the resolved commit sha
+        result=$(git for-each-ref "${unresolved_ref}" --format="%(refname)" --contains "${commit_sha}")
+        if [[ -z "${result}" ]]; then
+          cat << EOF > $(mktemp "$MINT_ERRORS/error-XXXX")
+      The \`meta-ref\` provided does not contain the resolved commit sha.
+      EOF
+          exit 1
+        fi
+      elif [[ "${{ params.ref }}" == "${commit_sha}" ]]; then
+        refs_containing_sha=$(git for-each-ref --format="%(refname)" --contains | grep -v refs/remotes)
+        unresolved_ref=$(echo "$refs_containing_sha" | head -n 1 | tr -d '\n')
```

Comment:
If the commit sha was provided as the ref, we can look for a branch/tag containing it

Reply:
does this only look for branches/tag where the sha is the head? if not, I think we probably should do that?

Reply:
I don't believe it does, I'll have to look into the git commands a little more to see which combination of incantations will get me there.

That said, I suspect we'll be likely to run into races around pushes in quick succession which will lead to this simply not being populated if we do that (honestly that's probably the case anyway, though, with force pushes)

Perhaps if it ends up being blank in those scenarios, we should set it to the SHA?

Reply:
> Perhaps if it ends up being blank in those scenarios, we should set it to the SHA?

Yeah, I think so.

Reply:
Done!